tests/ui_tools/test_boxes.py

```diff
@@ -13,8 +13,22 @@ def mock_external_classes(self, mocker, initial_index):
         self.view.model = mocker.Mock()
 
     @pytest.fixture()
-    def write_box(self, mocker):
+    def write_box(self, mocker, users_fixture, user_groups_fixture,
+                  streams_fixture, emojis_fixture):
         write_box = WriteBox(self.view)
+        write_box.view.users = users_fixture
+        write_box.model.user_group_names = [
+            groups['name'] for groups in user_groups_fixture]
+
+        write_box.view.pinned_streams = [
+            [stream['name']] for stream in
+            streams_fixture[:len(streams_fixture)//2]]
+        write_box.view.unpinned_streams = [
+            [stream['name']] for stream in
+            streams_fixture[len(streams_fixture)//2:]]
```

Comment:
Would it be enough to keep the elements for streams, mentions, and emojis in different fixtures? I would not prefer to keep things as they are - there would be considerable duplication in `test_generic_autocomplete_set_footer`. 
@preetmishra Would that work for your PR?

Reply:
@kaustubh-nair I would need a way to define what streams I want to be pinned (in parametrize) for every test case. Given the way we construct `pinned_streams`, it seems that we would need further refactoring just to accomplish that. I am thinking of _not_ using `streams_fixture` in https://github.com/zulip/zulip-terminal/pull/530#discussion_r403516773. Thoughts?

Reply:
Would it be a better idea to write a new set of test for testing pinned streams? The current streams test can be modified to not take pinning into consideration. This should decrease the fixture set but it will require very close inspection of the existing fixtures.

Reply:
@kaustubh-nair Could you explain how that would address my concern (that I mentioned in the previous comment)?

Reply:
Since the objective is to make pinned tests more explicit, one way to do it would be to test them seperately - this way you get more control over the way pinned streams are generated without affecting the current tests a lot. Does that make sense?
In any case, the `write_box` attributes for streams will require refactoring, so let me know what works best for you.

Reply:
Yes, "more control over the way pinned streams are generated" would be better. :+1: